README.md

```diff
@@ -1,41 +1,102 @@
+
+### `undeploy-*`
+
+#### Inputs
+
+| Name | Type | Description | Required | Default | Example |
+| ---- | ---- | ----------- | -------- | ------- | ------- |
+| `version-pattern` | string | A wildcard-supported string for version(s) of the model being removed | true | N/A | `2024.01.1-pre*` |
+
+#### Explanation
+
+This pipeline is responsible for removing deployed models from a deployment environment. Particular use-cases include removing `prerelease` builds from a deployment environment once a PR is merged.
+
+##### `undeploy-1-setup.yml`
+
+This workflow obtains the relevant spack and GitHub Environment information, and creates parallelized jobs removing the given `spack` environments based on the list of deployment environments.
+
+##### `undeploy-2-start.yml`
+
+Using the GitHub Environment, it `ssh`s into the deployment environments `spack` instance, and installs the model associated with the repository that called it. It then copies back relevant metadata and creates a versioned GitHub Release in the caller repository, if it is not a `prerelease` deployment.
+
+### `validate-json.yml`
+
+This workflow is used to validate the `config` folders `*.json` files based on their associated `*.schema.json`. This is used for PR checks on the `build-cd` repo itself.
+
+### `create-deployment-spack.yml`
+
+This workflow is used to create a version of `spack` on `Gadi`.
+
+#### Inputs
+
+| Name | Type | Description | Required | Default | Example |
+| ---- | ---- | ----------- | -------- | ------- | ------- |
+| `spack-version` | string | A version of `spack` | true | N/A | `0.20` |
+| `spack-packages-version` | string | A version of ACCESS-NRI/spack-packages to be bundled with the install of `spack` | true | `main` | `2023.11.12` |
+| `spack-config-version` | string | A version of ACCESS-NRI/spack-config to be bundled with the install of `spack` | true | `main` | `2024.01.01` |
+| `deployment-location` | true | A path in the deployment environment where Spack should be created. For example, if it is `opt/spack`, spack will be installed under `opt/spack/<spack-version>/` | true | N/A | `/opt/spack` |
 
 ## Usage
 
-For supported `spack`-installable ACCESS-NRI models, simply call the `deploy-1-setup.yml` reusable workflow from the given repositories workflow file, like so:
+### Deployment
+
+For supported `spack`-installable ACCESS-NRI models, simply call the `deploy-1-setup.yml` reusable workflow from the given repositories workflow file, as shown below. Don't forget to add required inputs!
 
 ```yml
 deploy:
   uses: access-nri/build-cd/.github/workflows/deploy-1-setup.yml@main
+  with:
+    type: release
+    version: 1.2.3
   secrets: inherit
   permissions:
     contents: write
 ```
```

Comment:
Consider moving this section up to the bottom of the `deploy-*` section. Then it is easier to see the required inputs.

Of the required inputs don't we imagine `type` will pretty much always be `release`? Won't the CI take care of making a `pre-release` deployment and then do a proper deployment on merge?

Reply:
Will do. And good shout on the `type` stuff. I just added it because it'd illustrate that it was an input, but that is probably better served by reading the `Inputs` section :D